### ISSUE TITLE
About switching host while space is opened

### ISSUE THREAD

**Issue by tinchodias** [pharo-graphics/Bloc#636]
In BlParallelUniverse, `closeSpaceSynchronously:` and `detachSpaceSynchronously:` share most part of code (we should reuse more) BUT ALSO, the diff smells to bug:

Left: `closeSpaceSynchronously:` - Right: `detachSpaceSynchronously:`
<img width="1664" alt="Screenshot 2024-10-22 at 13 49 42" src="https://github.com/user-attachments/assets/114f4097-572f-44f2-ad21-fba103415367">

When are they used?
- `closeSpaceSynchronously:`, when a space is closed.
- `detachSpaceSynchronously` when a space is opened and you change the host (like switch from a SDL2 window to a Morphic window on the fly, while it's open). This is not usual, so this is the candidate to be buggy (it should be adapt to what happens in close).

In fact, I didn't know this was implemented. I knew that switching host was working but when you `show`, then `close`, then change host, and send `show` again... I mean, when it is closed you can switch host right now.

Foe exmaple, this is implemented but buggy:
```smalltalk
space := BlSpace new.
space host: BlMorphicWindowHost new.
space root background: Color random.
space extent: 200 asPoint.
space show.

"wait, and evaluate:"
space host: BlOSWindowSDL2Host new.
```

**Comment by tinchodias** [comment]
Something similar happens in the counterpart operations: open and attach:

Left: `openSpaceSynchronously:` - Right: `attachSpaceSynchronously:`
<img width="1664" alt="Screenshot 2024-10-22 at 19 34 21" src="https://github.com/user-attachments/assets/f19a53a5-0fdf-4b7c-8c53-d3ae1d1c21ff">

It's error prone to have this semi-duplicated code, and `attachSpaceSynchronously:` is only used in the case if switching hosts.

**Comment by tinchodias** [comment]
+ I don't find traces of tests on this feature.

**Comment by tinchodias** [comment]
I added a note in a line added in `openSpaceSynchronously:` in PR #619

**Comment by tinchodias** [comment]
Fixed with [e2550e8](https://github.com/pharo-graphics/Bloc/commit/e2550e890470ebb7e867d85c0a27869d620b558d) + [47df523](https://github.com/pharo-graphics/Bloc/commit/47df523683ea4acac3c78b1d0cdbf94f25e98174) + [0a80636](https://github.com/pharo-graphics/Bloc/commit/0a80636dbd3139192c11275dbd8539ea515fb71c).

I'm missing an automatized test yet.
Tested manually with:
```smalltalk
space := BlSpace new.
space host: BlMorphicWindowHost new.
space root background: Color random.
space extent: 400 asPoint.
space title: 'hello'.
space show.

"wait, and evaluate:"
space host: BlOSWindowSDL2Host new.

"wait, and evaluate:"
space useMorphicHost 
```

https://github.com/user-attachments/assets/d400ce01-54a3-47a0-9535-482ffd2c7ce4
